test/model/store_test.dart

```diff
@@ -972,6 +992,74 @@ void main() {
     });
   });
 
+  group('UpdateMachine.poll reload failure', () {
+    late LoadingTestGlobalStore globalStore;
+
+    List<Completer<PerAccountStore>> completers() =>
+      globalStore.completers[eg.selfAccount.id]!;
+
+    Future<void> prepareReload(FakeAsync async) async {
+      globalStore = LoadingTestGlobalStore(accounts: [eg.selfAccount]);
+      final future = globalStore.perAccount(eg.selfAccount.id);
+      completers()
+        ..single.complete(
+            eg.store(globalStore: globalStore, account: eg.selfAccount))
+        ..clear();
+      final store = await future;
+      final updateMachine = globalStore.updateMachines[eg.selfAccount.id] =
+        UpdateMachine.fromInitialSnapshot(
+          store: store, initialSnapshot: eg.initialSnapshot());
+      updateMachine.debugPauseLoop();
+      updateMachine.poll();
+
+      (store.connection as FakeApiConnection).prepare(
+        apiException: eg.apiExceptionBadEventQueueId());
+      updateMachine.debugAdvanceLoop();
+      async.elapse(Duration.zero);
+      check(store).isLoading.isTrue();
+    }
+
+    void checkReloadFailure({
+      required FutureOr<void> Function() completeLoading,
+    }) {
+      awaitFakeAsync((async) async {
+        await prepareReload(async);
+
+        check(completers()).single.isCompleted.isFalse();
+        await completeLoading();
+        check(completers()).single.isCompleted.isTrue();
+        check(globalStore.takeDoRemoveAccountCalls()).single.equals(eg.selfAccount.id);
+
+        async.elapse(TestGlobalStore.removeAccountDuration);
+        check(globalStore.perAccountSync(eg.selfAccount.id)).isNull();
+
+        async.flushTimers();
+        // Reload never succeeds and there is no unhandled errors.
```

Comment:
nit: "errors" is plural, so "there are":
```suggestion
        // Reload never succeeds and there are no unhandled errors.
```